CNFE on scope defining annotations in the default package

**Describe the bug**
Quarkus fails ungracefully with scope-defining annotations on beans in the default package.
The error: https://gist.github.com/michalszynkiewicz/c489af90ed958b4ff43c7236f4a1b09a

**Expected behavior**
Either just work or produce a user-friendly message.

**To Reproduce**
Steps to reproduce the behavior:
1. clone https://github.com/michalszynkiewicz/quarkus-krakow/tree/failing-scope-on-default/ads
2. `mvn clean compile quarkus:dev` or `mvn clean package && java -jar target/*-runner.jar`

CC @manovotn

@gsmet good, point, here it is;
```
11:22:12,332 ERROR [io.qua.dev.DevModeMain] Failed to start quarkus: java.lang.ExceptionInInitializerError
        at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
        at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
        at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
        at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
        at java.lang.Class.newInstance(Class.java:442)
        at io.quarkus.runner.RuntimeRunner.run(RuntimeRunner.java:134)
        at io.quarkus.dev.DevModeMain.doStart(DevModeMain.java:180)
        at io.quarkus.dev.DevModeMain.start(DevModeMain.java:94)
        at io.quarkus.dev.DevModeMain.main(DevModeMain.java:66)
Caused by: java.lang.RuntimeException: Failed to start quarkus
        at io.quarkus.runner.ApplicationImpl1.<clinit>(ApplicationImpl1.zig:371)
        ... 9 more
Caused by: java.lang.NoClassDefFoundError: /AdsResource_ClientProxy
        at AdsResource_Bean.<init>(AdsResource_Bean.zig:130)
        at io.quarkus.arc.setup.Default_ComponentsProvider.getComponents(Default_ComponentsProvider.zig:219)
        at io.quarkus.arc.ArcContainerImpl.<init>(ArcContainerImpl.java:86)
        at io.quarkus.arc.Arc.initialize(Arc.java:18)
        at io.quarkus.arc.runtime.ArcRecorder.getContainer(ArcRecorder.java:33)
        at io.quarkus.deployment.steps.ArcProcessor$generateResources18.deploy_0(ArcProcessor$generateResources18.zig:53)
        at io.quarkus.deployment.steps.ArcProcessor$generateResources18.deploy(ArcProcessor$generateResources18.zig:369)
        at io.quarkus.runner.ApplicationImpl1.<clinit>(ApplicationImpl1.zig:317)
        ... 9 more
Caused by: java.lang.ClassNotFoundException: .AdsResource_ClientProxy
        at io.quarkus.runner.RuntimeClassLoader.findClass(RuntimeClassLoader.java:226)
        at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
        at io.quarkus.runner.RuntimeClassLoader.loadClass(RuntimeClassLoader.java:169)
        at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
        ... 17 more
```

--- a/independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/AbstractGenerator.java
+++ b/independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/AbstractGenerator.java
@@ -81,8 +81,8 @@ abstract class AbstractGenerator {
             }
         }
         String packageName = DotNames.packageName(providerTypeName);
-        if (packageName.startsWith("java.")) {
-            // It is not possible to place a class in a JDK package
+        if (packageName.isEmpty() || packageName.startsWith("java.")) {
+            // It is not possible to place a class in a JDK package or in default package
             packageName = DEFAULT_PACKAGE;
         }
         return packageName;

